torch/distributed/_tensor/placement_types.py

```diff
@@ -406,6 +406,16 @@ def __post_init__(self):
         if not isinstance(self.placements, tuple):
             self.placements = tuple(self.placements)
         self._hash: Optional[int] = None
+        if self.tensor_meta is not None:
+            size_symints = any(
+                isinstance(x, torch.SymInt) for x in self.tensor_meta.shape
```

Comment:
You can't hash SymInts unless you're willing to put a lot of guards on them.

Specifically, suppose you have `hash(s0) == hash(s1)`. If you hash SymInt using the sympy expression, this evaluates to False, but when the runtime values s0 == s1, then the expression should have been True.

Reply:
I'm a bit confusing here (or maybe lack of knowledge on sympy): if we have `hash(s0) == hash(s1)` then:

> If you hash SymInt using the sympy expression, this evaluates to False

How would this evaluates to False instead of true 🤔